src/stacks_events.rs

```diff
@@ -0,0 +1,86 @@
+use blockstack_lib::chainstate::stacks::StacksTransaction;
+use chrono::{DateTime, Local, SecondsFormat, Utc};
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde_json::{json, Value};
+use stacks_common::codec::StacksMessageCodec;
+use stacks_common::util::hash::hex_bytes;
+use std::fs::File;
+use std::io::prelude::*;
+use std::io::BufReader;
+use std::net::TcpListener;
+use std::net::TcpStream;
+use std::time::SystemTime;
+use std::{env, io};
+
+fn main() {
+    serve_for_events();
+}
+
+fn serve_for_events() {
+    let addr = "127.0.0.1:3700";
+    let listener = TcpListener::bind(addr).unwrap();
+    eprintln!("Listening on {}", addr);
+    for stream in listener.incoming() {
+        let mut stream = stream.unwrap();
+        handle_connection(stream);
+    }
+}
+
+lazy_static! {
+    static ref RE_POST: Regex = Regex::new(r"^POST /(.*?) HTTP/1.1\r\n$").unwrap();
+    static ref RE_CONTENT_LENGTH: Regex = Regex::new(r"^content-length: (\d+)\r\n$").unwrap();
+}
+
+fn handle_connection(mut stream: TcpStream) {
+    let mut buf = String::with_capacity(10 * 1024);
+    let mut reader = BufReader::new(stream.try_clone().unwrap());
+
+    let mut path = None;
+    let mut content_length = None;
+    let mut payload = None;
+
+    loop {
+        buf.clear();
+        reader.read_line(&mut buf).unwrap();
+        if path.is_none() {
+            let caps = RE_POST.captures(&buf).unwrap();
+            path = Some(caps.get(1).unwrap().as_str().to_string());
+        } else if content_length.is_none() {
+            let caps = RE_CONTENT_LENGTH.captures(&buf);
+            if let Some(caps) = caps {
+                content_length = Some(caps.get(1).unwrap().as_str().parse::<u64>().unwrap());
+            }
+        } else if buf.len() == 2 {
```

Comment:
Might want to explicitly check for `\r\n` since that's what it seems you're intending to do?